ivy/functional/ivy/general.py

```diff
@@ -524,6 +524,136 @@ def all_equal(
         Boolean, whether or not the inputs are equal, or matrix array of booleans if
         equality_matrix=True is set.
 
+    Functional Examples
```

Comment:
The header can simply be `Examples`

Reply:
Please also update the examples so that we have one example for each of `With Number inputs`, `With ivy.Array inputs`, `With ivy.NativeArray inputs` and so on. Also please make sure that one of the examples uses `equality_matrix=True`